ts/Series/FlowMap/FlowMapSeries.ts

```diff
@@ -0,0 +1,834 @@
+        if (growTowards) {
+            fromXToArc /= scaledWeight;
+            fromYToArc /= scaledWeight;
+            wX /= 4;
+            wY /= 4;
+        }
+
+        const shapeArgs: SVGAttributes = {
+            d: [
+                [
+                    'M',
+                    fromX - fromXToArc,
+                    fromY - fromYToArc
+                ],
+                [
+                    'Q',
+                    arcPointX - wX,
+                    arcPointY - wY,
+                    toX - toXToArc,
+                    toY - toYToArc
+                ],
+                // This is where markerEnd will be spliced to.
+                [
+                    'L',
+                    toX + toXToArc,
+                    toY + toYToArc
+                ],
+                [
+                    'Q',
+                    arcPointX + wX,
+                    arcPointY + wY,
+                    fromX + fromXToArc,
+                    fromY + fromYToArc
+                ],
+                ['Z']
+            ]
+        };
+
+        if (markerEndOptions && markerEndOptions.enabled && shapeArgs.d) {
+            const marker: SVGPath = FlowMapSeries.markerEndPath(
+                [toX - toXToArc, toY - toYToArc],
+                [toX + toXToArc, toY + toYToArc],
+                [toPoint.plotX as number, toPoint.plotY as number],
+                markerEndOptions
+            );
+
+            (shapeArgs.d as SVGPath).splice(2, 0,
+                ...marker
+            );
+        }
+
+        return shapeArgs;
+    }
+}
+
+/* *
+ *
+ *  Prototype properties
+ *
+ * */
+
+interface FlowMapSeries {
+    pointClass: typeof FlowMapPoint;
+}
+
+extend(FlowMapSeries.prototype, {
+    pointClass: FlowMapPoint,
+    // Make it work on zoom or pan
+    useMapGeometry: true
+});
+
+/* *
+ *
+ *  Registry
+ *
+ * */
+declare module '../../Core/Series/SeriesType' {
+    interface SeriesTypeRegistry {
+        flowmap: typeof FlowMapSeries;
+    }
+}
+
+SeriesRegistry.registerSeriesType('flowmap', FlowMapSeries);
+
+/* *
+ *
+ *  Default export
+ *
+ * */
+
+export default FlowMapSeries;
+
+/* *
+ *
+ *  API options
+ *
+ * */
+
+/**
+ * A `flowmap` series. If the [type](#series.flowmap.type) option
+ * is not specified, it is inherited from [chart.type](#chart.type).
+ *
+ * @extends   series,plotOptions.flowmap
+ * @excluding allAreas, allowPointSelect, boostBlending, boostThreshold,
+ * borderColor, borderWidth, dashStyle, dataLabels, dragDrop, joinBy, mapData,
+ * negativeColor, onPoint, shadow, showCheckbox
+ * @product   highmaps
+ * @apioption series.flowmap
+ */
+
+/**
+ * An array of data points for the series. For the `flowmap` series
+ * type, points can be given in the following ways:
+ *
+ * 1.  An array of arrays with 6 values. In this case, the values correspond
+ *     to `from, to, weight`. Example:
+ *     ```js
+ *     data: [
+ *         ["Point 1", "Point 2", 4],
+ *     ]
+ *     ```
+ *
```

Comment:
We talk about 6 values in the array, but only three are given. That might be a bit confusing for the users.